plugins/inputs/mqtt_consumer/mqtt_consumer.go

```diff
@@ -115,23 +118,41 @@ func (m *MQTTConsumer) Start(acc telegraf.Accumulator) error {
 		return fmt.Errorf("MQTT Consumer, invalid QoS value: %d", m.QoS)
 	}
 
+	if m.ConnectionTimeout <= 0 {
+		return fmt.Errorf("MQTT Consumer, invalid connection_timeout value: %d", m.ConnectionTimeout)
+	}
+
 	opts, err := m.createOpts()
 	if err != nil {
 		return err
 	}
 
 	m.client = mqtt.NewClient(opts)
+
+	log.Printf("%v", m.connect())
```

Comment:
Don't forget to remove this debugging line.

Reply:
I think this information might be helpful for users during their troubleshooting. What about adding debug variable into configuration and display this information only if it is true? 
```
func (m *MQTTConsumer) connect() error {
	if token := m.client.Connect(); token.Wait() && token.Error() != nil {

		err := token.Error()
		if m.debug {
			log.Printf("MQTT Consumer, connection error - %v", err)
		}
		return err
	}

	go m.receiver()

	return nil
}
```

Reply:
You can prefix your log message with `D!` and it will only be emitted if Telegraf is started with the debug flag.  In the future I'm hoping to switch to a logging system where the user can control the logging level of specific modules, but for now just use the global debug setting.